tests/cases/conformance/jsdoc/jsdocQuestionAddsOptionality.ts

```diff
@@ -0,0 +1,26 @@
+// @Filename: a.js
+// @noEmit: true
+// @allowJs: true
+// @checkJs: true
+// @strict: true
+/** @param {?number} a */
+function f(a) {
+    a = 1
+    a = null
+    a = undefined // should not be allowed
+}
+f()
+f(null)
+f(undefined) // should not be allowed
+f(1)
+
+/** @param {number?} a */
+function g(a) {
+    a = 1
+    a = null
+    a = undefined // should not be allowed
```

Comment:
If we remove the assignments and `return a;` presumably that should include `undefined`, because `a` might not have been provided?